### PR TITLE
Adds compression to server calls

### PR DESCRIPTION
**Pull request by fedefernandez**
Hi, thanks for making this library. I'm considering to integrate it in [Mu](https://github.com/higherkindness/mu/) and the compression at server level is the only feature I'm missing by the moment.

### CODE REVIEW

**Comment by fedefernandez** [comment]
@fiadliel @rossabaker @ahjohannessen feedback is welcomed, thanks!

**Comment by fedefernandez** [comment]
Thanks @ahjohannessen, @fiadliel @rossabaker anything on my own I need to do? ☺️

File: java-runtime/src/main/scala/server/models.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.lyranthe.fs2_grpc
+package java_runtime
+package server
+
+sealed abstract class ServerCompressor(val name: String) extends Product with Serializable
+case object GzipCompressor                               extends ServerCompressor("gzip")
+
+final case class ServerCallOptions(compressor: Option[ServerCompressor])
```
Comment (rossabaker) [inline java-runtime/src/main/scala/server/models.scala]: Do we anticipate having more options?  Case classes are challenging for maintaining binary compatibility, so I've started moving away from them for configuration.  If we do an abstract class, we can add more options later without breaking the interface.  Something like:

```scala
abstract class ServerCallOptions private (
  val compressor: Option[ServerCompressor]
) {
  def copy(compressor: Option[ServerCompressor] = this.compressor) =
    new ServerCallOptions(compressor) {}

  def withServerCompressor(compressor: Option[ServerCompressor]) =
    copy(compressor)
}

object ServerCallOptions {
  val default = new ServerCallOptions(None) {}
}
```

It's more verbose.  But when we add another one, we add to the constructor and the copy method, and the interface doesn't change.

**Comment by fedefernandez** [comment]
Oh great, I see, it's a nice tip. I'll address that change and let you know. 👍

**Comment by fedefernandez** [comment]
@rossabaker comments addressed

File: java-runtime/src/main/scala/server/models.scala [hunk 2]
```diff
@@ -5,8 +5,14 @@ package server
 sealed abstract class ServerCompressor(val name: String) extends Product with Serializable
 case object GzipCompressor                               extends ServerCompressor("gzip")
 
-final case class ServerCallOptions(compressor: Option[ServerCompressor])
+abstract class ServerCallOptions private (val compressor: Option[ServerCompressor]) {
+  def copy(compressor: Option[ServerCompressor] = this.compressor): ServerCallOptions =
+    new ServerCallOptions(compressor) {}
```
Comment (rossabaker) [inline java-runtime/src/main/scala/server/models.scala]: As these get more complicated, named arguments will be helpful here.  Especially if there are multiple arguments of the same type.  But with one, of course this is fine for now. :smile:

**Pull request merged**